### PR TITLE
Typehint opus.py

### PR DESCRIPTION
**Pull request by apple502j**
## Summary
Typehints opus.py. Checked on mypy.

- All native function calls are `#type: ignore`'d.
- `None` assignment inside destructors are `#type: ignore`'d.
- `Decoder.decode` now raises `InvalidArgument` instead of `OpusError` because `OpusError` cannot take a string as the error code. This is not technically breaking change, since it is unused and never documented.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/opus.py [hunk 1]
```diff
@@ -273,8 +280,8 @@ class OpusError(DiscordException):
         The error code returned.
     """
 
-    def __init__(self, code):
-        self.code = code
+    def __init__(self, code: int) -> None:
```
Comment (Rapptz) [inline discord/opus.py:283]:
```suggestion
    def __init__(self, code: int):
```

File: discord/opus.py [hunk 2]
```diff
@@ -300,92 +307,96 @@ def get_opus_version() -> str:
         return _lib.opus_get_version_string().decode('utf-8')
 
 class Encoder(_OpusStruct):
-    def __init__(self, application=APPLICATION_AUDIO):
+    def __init__(self, application: int=APPLICATION_AUDIO) -> None:
```
Comment (Rapptz) [inline discord/opus.py:310]:
```suggestion
    def __init__(self, application: int = APPLICATION_AUDIO):
```

File: discord/opus.py [hunk 3]
```diff
@@ -172,7 +179,7 @@ def _err_ne(result, func, args):
         [ctypes.c_char_p, ctypes.c_int], ctypes.c_int, _err_lt),
 ]
 
-def libopus_loader(name):
+def libopus_loader(name: str) -> ctypes.CDLL:
```
Comment (Rapptz) [inline discord/opus.py:182]:
```suggestion
def libopus_loader(name: str) -> Any:
```

Instead of `# type: ignore`ing everything this is the better way to go about doing it since the type checking is unnecessary.

File: discord/opus.py [hunk 4]
```diff
@@ -300,92 +307,96 @@ def get_opus_version() -> str:
...
-        ret = _lib.opus_encode(self._state, pcm, frame_size, data, max_data_bytes)
+        ret = _lib.opus_encode(self._state, pcm_ptr, frame_size, data, max_data_bytes) # type: ignore
 
-        return array.array('b', data[:ret]).tobytes()
+        # array can be initialized with bytes but mypy doesn't know
+        return array.array('b', data[:ret]).tobytes() # type: ignore
 
 class Decoder(_OpusStruct):
-    def __init__(self):
+    def __init__(self) -> None:
```
Comment (Rapptz) [inline discord/opus.py:369]:
```suggestion
    def __init__(self):
```

File: discord/opus.py [hunk 5]
```diff
@@ -395,28 +406,36 @@ def _set_gain(self, adjustment):
...
+        ...
+    
+    @overload
+    def decode(self, data: Literal[None], *, fec: Literal[False]) -> bytes:
+        ...
+
+    def decode(self, data: Optional[bytes], *, fec: bool = False) -> bytes:
         if data is None and fec:
-            raise OpusError("Invalid arguments: FEC cannot be used with null data")
+            raise InvalidArgument("Invalid arguments: FEC cannot be used with null data")
```
Comment (Rapptz) [inline discord/opus.py:438]: Why was this changed?

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/opus.py (hunk 5 above)
Comment (apple502j) [reply discord/opus.py:438]: `OpusError` takes `code: int`, this is `str`

**Review by Rapptz — APPROVED** [review]
LGTM.
